### PR TITLE
Add metrics package and docker compose for full-stack test

### PR DESCRIPTION
**Pull request by Sami1309**

### CODE REVIEW

File: metrics/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+WORKDIR /app
+
+COPY go.mod ./
+COPY go.sum ./
+RUN go mod download
+
+COPY *.go ./
+
+RUN go build -o /go-service
```
Comment (simba-git) [inline metrics/Dockerfile:11]: Minor but can we call it something that isn't this generic :)

File: metrics/interface.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+//generic interfaces exposed to the user
```
Comment (simba-git) [inline metrics/interface.go:14]: This comment isn't needed

File: metrics/interface.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	//remote packages
```
Comment (simba-git) [inline metrics/interface.go:4]: no need for this comment

File: metrics/main.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package main
```
Comment (simba-git) [inline metrics/main.go:1]: Let rename this file to metrics-example.go or something

File: metrics/prometheus/prometheus.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+scrape_configs:
+  - job_name: myapp
+    scrape_interval: 10s
+    metrics_path: /metrics
+    static_configs:
+      - targets: ["host.docker.internal:2112"]
```
Comment (simba-git) [inline metrics/prometheus/prometheus.yml:6]: What is this host? Is this using the docker network?

**Review by simba-git — COMMENTED** [review]
Few minor changes and questions

File: metrics/prometheus/prometheus.yml (hunk 5 above)
Comment (Sami1309) [reply metrics/prometheus/prometheus.yml]: It's syntax for how docker for windows exposes localhost. I'll add conventions for mac and linux (which just uses the normal localhost format)

**Pull request merged**